dexter/src/main/java/com/karumi/dexter/AndroidVersionPermissionMap.java

```diff
@@ -0,0 +1,185 @@
+            case Manifest.permission.RECEIVE_MMS:
+            case Manifest.permission.RECEIVE_SMS:
+            case Manifest.permission.RECEIVE_WAP_PUSH:
+            case Manifest.permission.RECORD_AUDIO:
+            case Manifest.permission.REORDER_TASKS:
+            case Manifest.permission.RESTART_PACKAGES:
+            case Manifest.permission.SEND_SMS:
+            case Manifest.permission.SET_ALWAYS_FINISH:
+            case Manifest.permission.SET_ANIMATION_SCALE:
+            case Manifest.permission.SET_DEBUG_APP:
+            case Manifest.permission.SET_PREFERRED_APPLICATIONS:
+            case Manifest.permission.SET_PROCESS_LIMIT:
+            case Manifest.permission.SET_TIME_ZONE:
+            case Manifest.permission.SET_WALLPAPER:
+            case Manifest.permission.SET_WALLPAPER_HINTS:
+            case Manifest.permission.SIGNAL_PERSISTENT_PROCESSES:
+            case Manifest.permission.STATUS_BAR:
+            case Manifest.permission.SYSTEM_ALERT_WINDOW:
+            case Manifest.permission.VIBRATE:
+            case Manifest.permission.WAKE_LOCK:
+            case Manifest.permission.WRITE_APN_SETTINGS:
+            case Manifest.permission.WRITE_CALENDAR:
+            case Manifest.permission.WRITE_CONTACTS:
+            case Manifest.permission.WRITE_GSERVICES:
+            case Manifest.permission.WRITE_SETTINGS:
+            case Manifest.permission.WRITE_SYNC_SETTINGS:
+                return true; //API 1
+            case Manifest.permission.BROADCAST_SMS:
+            case Manifest.permission.BROADCAST_WAP_PUSH:
+                return sdkVersion >= Build.VERSION_CODES.BASE_1_1;
+            case Manifest.permission.BIND_APPWIDGET:
+            case Manifest.permission.BIND_INPUT_METHOD:
+            case Manifest.permission.MOUNT_FORMAT_FILESYSTEMS:
+            case Manifest.permission.UPDATE_DEVICE_STATS:
+            case Manifest.permission.WRITE_SECURE_SETTINGS:
+                return sdkVersion >= Build.VERSION_CODES.CUPCAKE;
+            case Manifest.permission.CHANGE_WIFI_MULTICAST_STATE:
+            case Manifest.permission.GLOBAL_SEARCH:
+            case Manifest.permission.INSTALL_LOCATION_PROVIDER:
+            case Manifest.permission.WRITE_EXTERNAL_STORAGE:
+                return sdkVersion >= Build.VERSION_CODES.DONUT;
+            case Manifest.permission.ACCOUNT_MANAGER:
+                return sdkVersion >= Build.VERSION_CODES.ECLAIR;
+            case Manifest.permission.BIND_DEVICE_ADMIN:
+            case Manifest.permission.BIND_WALLPAPER:
+            case Manifest.permission.KILL_BACKGROUND_PROCESSES:
+            case Manifest.permission.SET_TIME:
+                return sdkVersion >= Build.VERSION_CODES.FROYO;
+            case Manifest.permission.NFC:
+            case Manifest.permission.SET_ALARM:
+            case Manifest.permission.USE_SIP:
+                return sdkVersion >= Build.VERSION_CODES.GINGERBREAD;
```

Comment:
We just need to support API 10 => so we should remove all the code checking permissions before ``Build.VERSION_CODES.GINGERBREAD``